openshift-kube-apiserver/admission/customresourcevalidation/apirequestcount/validate_apirequestcount.go

```diff
@@ -0,0 +1,112 @@
+package apirequestcount
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	apiv1 "github.com/openshift/api/apiserver/v1"
+	"k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/kubernetes/openshift-kube-apiserver/admission/customresourcevalidation"
+)
+
+const PluginName = "config.openshift.io/ValidateAPIRequestCount"
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(config io.Reader) (admission.Interface, error) {
+		return NewValidateAPIRequestCount()
+	})
+}
+
+func NewValidateAPIRequestCount() (admission.Interface, error) {
+	return customresourcevalidation.NewValidator(
+		map[schema.GroupResource]bool{
+			apiv1.GroupVersion.WithResource("apirequestcounts").GroupResource(): true,
+		},
+		map[schema.GroupVersionKind]customresourcevalidation.ObjectValidator{
+			apiv1.GroupVersion.WithKind("APIRequestCount"): apiRequestCountV1{},
+		})
+}
+
+type apiRequestCountV1 struct {
+}
+
+func toAPIRequestCountV1(uncastObj runtime.Object) (*apiv1.APIRequestCount, field.ErrorList) {
+	if uncastObj == nil {
+		return nil, nil
+	}
+
+	obj, ok := uncastObj.(*apiv1.APIRequestCount)
+	if !ok {
+		return nil, field.ErrorList{
+			field.NotSupported(field.NewPath("kind"), fmt.Sprintf("%T", uncastObj), []string{"APIRequestCount"}),
+			field.NotSupported(field.NewPath("apiVersion"), fmt.Sprintf("%T", uncastObj), []string{"apiserver.openshift.io/v1"}),
+		}
+	}
+
+	return obj, nil
+}
+
+func (a apiRequestCountV1) ValidateCreate(uncastObj runtime.Object) field.ErrorList {
+	obj, errs := toAPIRequestCountV1(uncastObj)
```

Comment:
obj might be `nil`, right? in that case it will blow up in line `60`

Reply:
Should not happen. If it is `nil`, it is worthy of a `panic`. All other CRD validators assume not `nil` also.

Reply:
okay, I saw a check in `toAPIRequestCountV1` func:
```
if uncastObj == nil {
		return nil, nil
	}
```

Reply:
Updated. I removed the `nil` check in `toAPIRequestCountV1()`.